app/lib/package.rb

```diff
@@ -62,17 +70,39 @@ def add_pages(zip_file, manifest)
     end
   end
 
+  def add_tips(zip_file, manifest)
+    @tips.each do |tip|
```

Comment:
a tip can be referenced from multiple locations in the xml, right now you aren't filtering unique tips. This behavior is similar to how `@resources` includes unique resources